staging/src/k8s.io/kube-aggregator/pkg/controllers/autoregister/autoregister_controller.go

```diff
@@ -253,7 +258,13 @@ func (c *autoRegisterController) checkAPIService(name string) (err error) {
 
 	// we have a spurious APIService that we're managing, delete it (5A,6A)
 	case desired == nil:
-		return c.apiServiceClient.APIServices().Delete(curr.Name, nil)
+		opts := &metav1.DeleteOptions{Preconditions: metav1.NewUIDPreconditions(string(curr.UID))}
+		err := c.apiServiceClient.APIServices().Delete(curr.Name, opts)
```

Comment:
i was doubting the unconditional deletion here is the cause of flooding error reported in this thread previously [https://github.com/kubernetes/kubernetes/issues/70287](https://github.com/kubernetes/kubernetes/issues/70287). ocassionally, we might hit similar problem where the aggregators in HA setup is racing between creating/deleting apiservices. the same apiservice keeps being created by one aggregator but instantly deleted by another aggregator. so even an invalid apiservice might be popped from backoff queuing if it's ephemeral one.  (am not really sure so i didn't submit this change w/ a conclusion)

[https://github.com/kubernetes/kubernetes/blob/3478647333c91689cf4c737012a60e6d70a661e7/staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go#L145-L147](https://github.com/kubernetes/kubernetes/blob/3478647333c91689cf4c737012a60e6d70a661e7/staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go#L145-L147)